modules/geo-layers/src/wms-layer/wms-layer.ts

```diff
@@ -28,19 +28,23 @@ type _WMSLayerProps = {
   data: string | ImageSource;
   serviceType?: ImageServiceType | 'auto';
   layers?: string[];
-  srs?: 'EPSG:4326' | 'EPSG:3857' | 'auto';
+  crs?: 'EPSG:4326' | 'EPSG:3857' | 'auto';
   onMetadataLoad?: (metadata: ImageSourceMetadata) => void;
   onMetadataLoadError?: (error: Error) => void;
   onImageLoadStart?: (requestId: unknown) => void;
   onImageLoad?: (requestId: unknown) => void;
   onImageLoadError?: (requestId: unknown, error: Error) => void;
+
+  /** @deprecated Use props.crs instead */
+  srs?: 'EPSG:4326' | 'EPSG:3857' | 'auto';
 };
 
 const defaultProps: DefaultProps<WMSLayerProps> = {
   id: 'imagery-layer',
   data: '',
   serviceType: 'auto',
-  srs: 'auto',
+  crs: undefined, // 'auto' We assign the default manually to support the deprecated srs prop
+  srs: undefined,
```

Comment:
I don't see any benefit of explicitly defining them this way, for functionality nor readability. The purpose of `defaultProps` is to define fallback values and validation/comparison behaviors. If you don't want any you can just leave these props out.

There is also [deprecatedFor](https://deck.gl/docs/developer-guide/custom-layers/prop-types#property-types-1) that will handle deprecation automatically. But we may not want the warning if this is considered a silent API change.

Reply:
The problems is that if `props.crs` is not defined, then I want to use props.srs. However if we provide default values then I can't detect if if props.crs was supplied or not. I.e. it could have been explicitly specified as `auto`.

I did feel this was a case of silent deprecation. I felt that this is an experimental layer and not worth making a big deal of the initial round of deprecations, just tried to avoid breaking early adopters by still supporting srs.

FWIW, the `@deprecated` tag visually marks the field with ~~strikethrough~~ at least in my editor. Also makes it easy to search for deprecated props once the time comes for a major release.

Reply:
Then you can just delete these two lines?

Most users don't use the typed endpoint as it's still experimental.

Reply:
Matter of style and taste I guess. I explicitly list out all props in similar code I control, to show that `undefined` is intentional and not by omission. But happy to follow your guidelines in deck.